pyrenew_hew/ww_site_level_dynamics_model.py

```diff
@@ -0,0 +1,538 @@
+                [
+                    initial_exp_growth_rate_ref_subpop,
+                    initial_exp_growth_rate_non_ref_subpop,
+                ]
+            )
+
+            rtu_subpop_ar_proc = ARProcess()
+            rtu_subpop_ar_weekly = rtu_subpop_ar_proc(
+                noise_name="rtu_ar_proc",
+                n=n_weeks_post_init,
+                init_vals=rtu_subpop_ar_init[jnp.newaxis],
+                autoreg=autoreg_rt_subpop[jnp.newaxis],
+                noise_sd=sigma_rt,
+            )
+            numpyro.deterministic("rtu_subpop_ar_weekly", rtu_subpop_ar_weekly)
+            log_rtu_non_ref_subpop_in_week = (
+                rtu_subpop_ar_weekly + log_r_t_in_weeks[:, jnp.newaxis]
+            )
+            log_rtu_subpop_in_week = jnp.concat(
+                [
+                    log_rtu_ref_subpop_in_week[:, jnp.newaxis],
+                    log_rtu_non_ref_subpop_in_week,
+                ],
+                axis=1,
+            )
+
+        numpyro.deterministic("i_first_obs_over_n_subpop", i_first_obs_over_n_subpop)
+        numpyro.deterministic(
+            "initial_exp_growth_rate_subpop", initial_exp_growth_rate_subpop
+        )
+
+        log_i0_subpop = (
+            jnp.log(i_first_obs_over_n_subpop)
+            - self.unobs_time * initial_exp_growth_rate_subpop
+        )
+        numpyro.deterministic("log_i0_subpop", log_i0_subpop)
+
+        rtu_subpop = jnp.squeeze(
+            jnp.repeat(
+                jnp.exp(log_rtu_subpop_in_week),
+                repeats=7,
+                axis=0,
+            )[:n_datapoints, :]
+        )
+        numpyro.deterministic("rtu_subpop", rtu_subpop)
+
+        i0_subpop_rv = DeterministicVariable("i0_subpop", jnp.exp(log_i0_subpop))
+        initial_exp_growth_rate_subpop_rv = DeterministicVariable(
+            "initial_exp_growth_rate_subpop", initial_exp_growth_rate_subpop
+        )
+
+        infection_initialization_process = InfectionInitializationProcess(
+            "I0_initialization",
+            i0_subpop_rv,
+            InitializeInfectionsExponentialGrowth(
+                self.n_initialization_points,
+                initial_exp_growth_rate_subpop_rv,
+                t_pre_init=self.i0_t_offset,
+            ),
+        )
+
+        generation_interval_pmf = self.generation_interval_pmf_rv()
+        i0 = infection_initialization_process()
+        numpyro.deterministic("i0", i0)
+
+        inf_with_feedback_proc_sample = self.inf_with_feedback_proc.sample(
+            Rt=rtu_subpop,
+            I0=i0,
+            gen_int=generation_interval_pmf,
+        )
+
+        new_i_subpop = jnp.atleast_2d(
+            jnp.concat(
+                [
+                    i0,
+                    inf_with_feedback_proc_sample.post_initialization_infections,
+                ]
+            )
+        )
+        if new_i_subpop.shape[0] == 1:
+            new_i_subpop = new_i_subpop.T
```

Comment:
Can we do something different in the creation of `new_i_subpop` to avoid this?

Reply:
I can't think of alternative implementation that works with both 1D and 2D `new_i_subpop` but happy to implement if you have ideas.

Reply:
Would `expand_dims` do this?